### PR TITLE
Add 'BulkThumbnailCreator' project

### PR DESCRIPTION
**Pull request by BugMixer95**
This PR contains new application called **Bulk Thumbnail Creator**.
The main purpose of it is to allow user to resize/rename a huge amount of images.
The application also write log entries to Windows Event Viewer and to local file.

### CODE REVIEW

File: Shastak/BulkThumbnailCreator/Extras/BTCLogger.cs [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+    {
+        private static string LogsFilename { get; set; }
+
+        public static void WriteLocalLogs(string message)
+        {
+            CreateLogsFileForToday();
+
+            using (StreamWriter sw = new StreamWriter(LogsFilename, true))
+            {
+                string now = DateTime.Now.ToString("yyyy-MM-dd HH:mm:ss.fffff");
```
Comment (Audyman) [inline Shastak/BulkThumbnailCreator/Extras/BTCLogger.cs]: Better to move date format to config or to separate class for string constants

File: Shastak/BulkThumbnailCreator/Extras/BTCLogger.cs [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+                sw.WriteLine($"[{now}] {message}.");
+            }
+        }
+
+        private static void CreateLogsFileForToday()
+        {
+            string today = DateTime.Now.ToString("yyyy-MM-dd");
```
Comment (Audyman) [inline Shastak/BulkThumbnailCreator/Extras/BTCLogger.cs]: see above

File: Shastak/BulkThumbnailCreator/Extras/BTCLogger.cs [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+            string filename = @$"{DirectoryHandler.Logs}\{today}_logs.txt";
+
+            if (!File.Exists(filename))
+            {
+                using (StreamWriter sw = new StreamWriter(filename, true))
+                {
+                    string now = DateTime.Now.ToString("yyyy-MM-dd HH:mm:ss.fffff");
```
Comment (Audyman) [inline Shastak/BulkThumbnailCreator/Extras/BTCLogger.cs]: see above

File: Shastak/BulkThumbnailCreator/Modules/ImageHandler.cs [hunk 4]
```diff
@@ -0,0 +1,229 @@
...
+
+                    DisplayOperationResult(isOperationAborted, $"Operation was aborted by {Environment.UserName}.", processedFiles, skippedFiles);
+
+                    return;
+                }
+
+                // if image is successfully processed it goes to the processed files directory
+                try
+                {
+                    BTCLogger.WriteLocalLogs($"Starting resizing file {Files[i].Name}");
```
Comment (Audyman) [inline Shastak/BulkThumbnailCreator/Modules/ImageHandler.cs]: naming: BtcLogger

File: Shastak/BulkThumbnailCreator/Modules/ImageHandler.cs [hunk 5]
```diff
@@ -0,0 +1,229 @@
...
+            DirectoryInfo folder = new DirectoryInfo(DirectoryHandler.SourceFiles);
+            Files = folder.GetFiles();
+            int filesCount = Files.Length;
+
+            // if there are no files in the source folder show a warning message for user to add files
+            while (filesCount == 0)
+            {
+                TextColorizer.WriteTextInColor(
+                    "Your source folder doesn't contain any files! \n" +
+                    "Please add some files in the source folder and press any key to continue. \n",
```
Comment (Audyman) [inline Shastak/BulkThumbnailCreator/Modules/ImageHandler.cs]: Better to move message texts to separate class e.g. Constants

File: Shastak/BulkThumbnailCreator/Program.cs [hunk 6]
```diff
@@ -0,0 +1,130 @@
...
+            }
+        }
+
+        private static void SayHello()
+        {
+            Console.Clear();
+
+            Console.Write("Welcome to the ");
+            TextColorizer.WriteTextInColor("BulkThumbnailCreator", ConsoleColor.Green, false);
+            Console.WriteLine("! \nThis program allows you to resize or rename your images.");
```
Comment (Audyman) [inline Shastak/BulkThumbnailCreator/Program.cs]: see above

**Review by Audyman — APPROVED** [review]
some minor comments, but code base is very good

File: Shastak/BulkThumbnailCreator/Modules/ImageHandler.cs (hunk 4 above)
Comment (BugMixer95) [reply Shastak/BulkThumbnailCreator/Modules/ImageHandler.cs]: fixed

File: Shastak/BulkThumbnailCreator/Extras/BTCLogger.cs (hunk 1 above)
Comment (BugMixer95) [reply Shastak/BulkThumbnailCreator/Extras/BTCLogger.cs]: fixed by adding Constants.cs class

File: Shastak/BulkThumbnailCreator/Extras/BTCLogger.cs (hunk 2 above)
Comment (BugMixer95) [reply Shastak/BulkThumbnailCreator/Extras/BTCLogger.cs]: fixed by adding Constants.cs class

File: Shastak/BulkThumbnailCreator/Extras/BTCLogger.cs (hunk 3 above)
Comment (BugMixer95) [reply Shastak/BulkThumbnailCreator/Extras/BTCLogger.cs]: fixed by adding Constants.cs class

File: Shastak/BulkThumbnailCreator/Modules/ImageHandler.cs (hunk 5 above)
Comment (BugMixer95) [reply Shastak/BulkThumbnailCreator/Modules/ImageHandler.cs]: fixed by adding Constants.cs class

File: Shastak/BulkThumbnailCreator/Program.cs (hunk 6 above)
Comment (BugMixer95) [reply Shastak/BulkThumbnailCreator/Program.cs]: fixed by adding Constants.cs class

**Pull request merged**